src/EcoSurveyComponent.java

```diff
@@ -0,0 +1,147 @@
+import components.map.Map;
+import components.map.Map1L;
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+public class EcoSurveyComponent {
+    private final class Species {
+
+        //Should these fields be public?
+        /**
+         * colloquial name
+         */
+        private String name;
+        /**
+         * scientific name
+         */
+        private String trueName;
+        /**
+         * ideal population density
+         */
+        private double density;
+
+    }
+
+    //is this redundant. Should I just create another private field for count?
+    private Map<Species, Integer> speciesCount;
+
+    private double area;
+
+    private int totalSpecies;
```

Comment:
Unless this is meant to be an upper bound, I wouldn't store this. You can get the total from the length of the map. If this is meant to store the total number of creatures, this variable might be useful for saving you computation time.